src/test/peerman_tests.cpp

```diff
@@ -0,0 +1,307 @@
+#include <test/util/setup_common.h>
+#include <util/string.h>
+#include <util/system.h>
+#include <util/time.h>
+#include <timedata.h>
+#include <validation.h>
+
+#include <array>
+#include <optional>
+#include <stdint.h>
+
+#include <boost/test/unit_test.hpp>
+
+static CService ip(uint32_t i)
+{
+    struct in_addr s;
+    s.s_addr = i;
+    return CService(CNetAddr(s), Params().GetDefaultPort());
+}
+
+static void AdvanceMockTime(int64_t delta)
+{
+    auto time_before{GetMockTime()};
+    SetMockTime(time_before + std::chrono::seconds(delta));
+}
+
+/** Mock the connections interface. */
+class ConnectionsInterfaceMock : public ConnectionsInterface
+{
+public:
+    bool ForNode(NodeId id, std::function<bool(CNode* pnode)> func) override { return true; };
+    using NodeFn = std::function<void(CNode*)>;
+    void ForEachNode(const NodeFn& func) override {};
+    void ForEachNode(const NodeFn& func) const override {};
+    void PushMessage(CNode* pnode, CSerializedNetMsg&& msg) override;
+    CSipHasher GetDeterministicRandomizer(uint64_t id) const override { return {0, 0}; };
+    void WakeMessageHandler() override {};
+    bool OutboundTargetReached(bool historicalBlockServingLimit) const override { return true; };
+    std::vector<CAddress> GetAddresses(size_t max_addresses,
+                                       size_t max_pct,
+                                       std::optional<Network> network) const override { return {}; };
+    std::vector<CAddress> GetAddresses(CNode& requestor, size_t max_addresses, size_t max_pct) override { return {}; };
+    bool DisconnectNode(const CNetAddr& addr) override { return true; };
+    unsigned int GetReceiveFloodSize() const override { return 0; };
+    int GetExtraFullOutboundCount() const override { return 0; };
+    int GetExtraBlockRelayCount() const override { return 0; };
+    void SetTryNewOutboundPeer(bool flag) override {};
+    bool GetTryNewOutboundPeer() const override { return true; };
+    bool GetNetworkActive() const override { return true; };
+    bool GetUseAddrmanOutgoing() const override { return true; };
+    void StartExtraBlockRelayPeers() override {};
+    bool ShouldRunInactivityChecks(const CNode& node, std::chrono::seconds now) const override { return true; };
+    bool CheckIncomingNonce(uint64_t nonce) override { return true; };
+
+    void NodeReceiveMsgBytes(CNode& node, Span<const uint8_t> msg_bytes, bool& complete) const;
+    bool ReceiveMsgFrom(CNode& node, CSerializedNetMsg& ser_msg) const;
+    void OnPing(CDataStream& data);
+
+    virtual ~ConnectionsInterfaceMock() {}
+
+    /** Count of number of each message type sent */
+    std::map<std::string, uint64_t> m_message_types_sent;
+    /** Most recent ping nonce received. */
+    uint64_t m_ping_nonce{0};
+
+};
+
+void ConnectionsInterfaceMock::PushMessage(CNode* pnode, CSerializedNetMsg&& msg)
+{
+    BOOST_TEST_MESSAGE(strprintf("received message %s from peer %d", msg.m_type, pnode->GetId()));
```

Comment:
Maybe I am missing something, but shouldn't this be `sending message %s to peer %d`??

Reply:
Right, `PushMessage` is called to send a message to a peer. Done.